### ISSUE TITLE
Segfault when creating single-argument events

### ISSUE THREAD

**Issue by mavam** [bro/broker#11]
I'm experimenting with Bro to Python communication and run into a scenario where I get segfault. Here are the Bro and Python scripts:

```bro
redef exit_only_after_terminate = T;

event foo()
  {
  # nop
  }

event bar(x: int)
  {
  print x;
  }

event Broker::peer_added(endpoint: Broker::EndpointInfo, msg: string)
  {
  local e = Broker::make_event(foo);
  Broker::publish("/test", e);
  }

event Broker::peer_lost(endpoint: Broker::EndpointInfo, msg: string)
  {
  terminate();
  }

event bro_init()
  {
  Broker::subscribe("/test");
  Broker::peer("localhost", 55555/tcp);
  }
```

```python
#!/usr/bin/env python

import broker
import broker.bro

# Setup endpoint and subscribers.
endpoint = broker.Endpoint()
endpoint.listen("localhost", 55555)
subscriber = endpoint.make_subscriber("/test")
while True:
    print("waiting for remote signal...")
    topic, data = subscriber.get()
    print("got message:", data)
    event = broker.bro.Event("bar", 42)    # <-------------------- crash
    endpoint.publish("/test", event)
```

When I run first the Python (`python -X faulthandler test.py`) and then the Bro script, I get a backtrace that points to this function:

```python
class Event(_broker.bro.Event):
    def __init__(self, *args):
        if len(args) == 1 and isinstance(args[0], list):
            # Parse raw broker message as event.
            _broker.bro.Event.__init__(self, broker.Data.from_py(args[0]))
        elif len(args) == 2 and isinstance(args[0], str) and not isinstance(args[0], broker.Data):
            # (name, (arg1, arg2, ...))
            _broker.bro.Event.__init__(self, args[0], broker.Data.from_py(args[1])) # <-------------- crash
        else:
            # (name, arg1, arg2, ...)
            _broker.bro.Event.__init__(self, args[0], broker.Data.from_py(args[1:]))

    def args(self):
        return [broker.Data.to_py(a) for a in _broker.bro.Event.args(self)]
```

The segfault doesn't show up when I add a second argument (e.g.l `broker.bro.Event("bar", 42, 42)`).

I'm not sure what's going on here under the hood yet and how to use the machinery. For example, why do we need `broker.Data.from_py(x)` when we can write `broker.Data(x)`?

**Comment by jsiwek** [comment]
>         elif len(args) == 2 and isinstance(args[0], str) and not isinstance(args[0], broker.Data):
>             # (name, (arg1, arg2, ...))
>             _broker.bro.Event.__init__(self, args[0], broker.Data.from_py(args[1])) # <-------------- crash
>         else:
>             # (name, arg1, arg2, ...)
>             _broker.bro.Event.__init__(self, args[0], broker.Data.from_py(args[1:]))

Thanks.  Should be fixed via 054eac2.

> I'm not sure what's going on here under the hood yet and how to use the machinery. For example, why do we need broker.Data.from_py(x) when we can write broker.Data(x)?

My guess is we don't *need* to use `from_py(x)` instead of  `Data(x)` since the former just calls the later.  If the question is rather "why do both exist?", I don't know.  Maybe a static method was needed for something.

**Comment by mavam** [comment]
> Thanks. Should be fixed via 054eac2.

Awesome, thanks!
